sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumbersClient/PhoneNumbersClientLiveTests.cs

```diff
@@ -312,6 +312,95 @@ public void CreateSearch()
             Assert.AreEqual(PhoneNumberType.TollFree, response.Value.PhoneNumberType);
         }
 
+        [Test]
+        [SyncOnly]
+        public void CreateSearchPurchaseReleaseDNR()
+        {
+            if (TestEnvironment.ShouldIgnorePhoneNumbersTests)
+            {
+                Assert.Ignore("Skip phone number live tests flag is on.");
+            }
+            var client = CreateClient();
+            var searchOperation = client.StartSearchAvailablePhoneNumbers("US", PhoneNumberType.TollFree, PhoneNumberAssignmentType.Application,
+                new PhoneNumberCapabilities(PhoneNumberCapabilityType.Outbound, PhoneNumberCapabilityType.None));
+
+            while (!searchOperation.HasCompleted)
+            {
+                SleepIfNotInPlaybackMode();
+                searchOperation.UpdateStatus();
+            }
+
+            Assert.IsTrue(searchOperation.HasCompleted);
+            Assert.AreEqual(1, searchOperation.Value.PhoneNumbers.Count);
+            Assert.AreEqual(PhoneNumberAssignmentType.Application, searchOperation.Value.AssignmentType);
+            Assert.AreEqual(PhoneNumberCapabilityType.Outbound, searchOperation.Value.Capabilities.Calling);
+            Assert.AreEqual(PhoneNumberCapabilityType.None, searchOperation.Value.Capabilities.Sms);
+            Assert.AreEqual(PhoneNumberType.TollFree, searchOperation.Value.PhoneNumberType);
+
+            var phoneNumber = searchOperation.Value.PhoneNumbers.Single();
+            var searchId = searchOperation.Value.SearchId;
+            var consentToNotResellNumbers = true;
+
+            var purchaseOperation = client.StartPurchasePhoneNumbers(searchId, consentToNotResellNumbers);
+
+            while (!purchaseOperation.HasCompleted)
+            {
+                SleepIfNotInPlaybackMode();
+                purchaseOperation.UpdateStatus();
+            }
+
+            Assert.AreEqual(purchaseOperation.GetRawResponse().Status, 200);
+
+            var releaseOperation = client.StartReleasePhoneNumber(phoneNumber);
+
+            while (!releaseOperation.HasCompleted)
+            {
+                SleepIfNotInPlaybackMode();
+                releaseOperation.UpdateStatus();
+            }
+
+            Assert.IsTrue(releaseOperation.HasCompleted);
+        }
+
+        [Test]
+        [SyncOnly]
+        public void CreateSearchPurchaseFailureDNR()
+        {
+            if (TestEnvironment.ShouldIgnorePhoneNumbersTests)
+            {
+                Assert.Ignore("Skip phone number live tests flag is on.");
+            }
+            var client = CreateClient();
+            var searchOperation = client.StartSearchAvailablePhoneNumbers("US", PhoneNumberType.TollFree, PhoneNumberAssignmentType.Application,
```

Comment:
I don't understand. Why is US subject to DNR?

Reply:
This was done for testing purposes as currently DNR countries are manual acquisition only.
Also the same reason I have added them in the tests, I will add a comment to change it to any of the DNR countries once we have automatic acquisition.